migrations/v2.js

```diff
@@ -0,0 +1,205 @@
+   */
+  mutateContent('adapt-contrib-accordion - add accordion._supportedLayout', async () => {
+    accordions.forEach(accordion => {
+      accordion._supportedLayout = 'half-width';
+    });
+    return true;
+  });
+
+  checkContent('adapt-contrib-accordion - check accordion._supportedLayout attribute', async () => {
+    const isValid = accordions.every(({ _supportedLayout }) => _supportedLayout !== undefined);
+    if (!isValid) throw new Error('adapt-contrib-accordion - _supportedLayout not added to every instance of accordion');
+    return true;
+  });
+
+  /**
+   * * Add JSON field to component and set blank.
+   */
+  mutateContent('adapt-contrib-accordion - add accordion.instruction', async () => {
+    accordions.forEach(accordion => {
+      accordion.instruction = '';
+    });
+    return true;
+  });
+
+  checkContent('adapt-contrib-accordion - check accordion.instruction atrribute', async () => {
+    const isValid = accordions.every(({ instruction }) => instruction !== undefined);
+    if (!isValid) throw new Error('adapt-contrib-accordion - instruction not added to every instance of accordion');
+    return true;
+  });
+
+  /**
+   * * Add field to each item in a JSON array and set blank.
+   */
+  mutateContent('adapt-contrib-accordion - add accordion._items._graphic.alt attribute', async () => {
+    accordions.forEach(accordion => {
+      accordion._items.forEach(item => {
+        item._graphic.alt = '';
+      });
+    });
+    return true;
+  });
+
+  checkContent('adapt-contrib-accordion - check add accordion._items._graphic.alt atrribute', async () => {
+    const isValid = accordions.every(accordion =>
+      accordion._items.every(item =>
+        item._graphic && item._graphic.alt !== undefined
+      )
+    );
+    if (!isValid) throw new Error('adapt-contrib-accordion - _graphic.alt not added to every instance of accordion._items');
+    return true;
+  });
+
+  /**
+   * * Add field to each item in a JSON array and set blank.
+   */
+  mutateContent('adapt-contrib-accordion - add accordion._items._graphic.src attribute', async () => {
+    accordions.forEach(accordion => {
+      accordion._items.forEach(item => {
+        item._graphic.src = '';
+      });
+    });
+    return true;
+  });
+
+  checkContent('adapt-contrib-accordion - check add accordion._items._graphic.src atrribute', async () => {
+    const isValid = accordions.every(accordion =>
+      accordion._items.every(item =>
+        item._graphic && item._graphic.src !== undefined
+      )
+    );
+    if (!isValid) throw new Error('adapt-contrib-accordion - _graphic.src not added to every instance of accordion._items');
+    return true;
+  });
+
+  /**
+    * * Add a new field to course globals.
+    */
+  mutateContent('adapt-contrib-accordion - add globals ariaRegion attribute', async (content) => {
+    course = content.find(({ _type }) => _type === 'course');
+    courseAccordionGlobals = course._globals._components._accordion ?? {};
```

Comment:
If course._globals._components._accordion doesn't exist this is setting a variable to {} and isn't putting that back into content. Dan has used a helper function and Ollie has suggested using the lodash library, so something like below would check if the nested path exists and create it if not. 

```suggestion
if (!_.has(course, '_globals._components._accordion')) _.set(course, '_globals._components._accordion', {});
    courseAccordionGlobals = course._globals._components._accordion;
```

This could then be copied to where ever you are searching for globals. Also need to add `import _ from 'lodash'` at the top of each file.

Reply:
Much appreciated.